common/Unit.cpp

```diff
@@ -498,10 +498,16 @@ void UnitBase::exitTest(TestResult result, const std::string& reason)
     }
 }
 
+std::string UnitBase::getReason() const
+{
+    std::lock_guard<std::mutex> guard(_lock);
```

Comment:
Nit: sooner or later it would be nice to use something like https://github.com/dragazo/rustex to make sure you can't forget the lock the data you are supposed to read when the intent is to always lock first. But this is fine, it's how the surrounding code does it everywhere in the code so far.